core/src/main/resources/assets/francois/templates.ts

```diff
@@ -157,4 +189,89 @@ export class CreateJob {
         console.log(evt.target.value);
         this.newJobName = evt.target.value;
     }
-}
\ No newline at end of file
+}
+
+
+@Component({
+    templateUrl: Consts.basePath + '/job-edit.html',
+    directives: [FORM_DIRECTIVES, TopNav, ROUTER_DIRECTIVES, NgFor, NgIf, CORE_DIRECTIVES],
+})
+export class EditJob {
+
+    public parameters:any[] = [];
+
+    public templateName:string;
+
+    public jobName:string;
+
+    public isCreateable:boolean = false;
+    public createSucceeded:boolean = false;
+
+    constructor(private francoisApi:FrancoisApi,
+                private routeParams:RouteParams) {
+
+        this.templateName = routeParams.get('templateName');
+        this.jobName = routeParams.get('jobName');
+
+        francoisApi.getTemplateParameters(this.templateName)
+            .map(r => r.json())
+            .subscribe(template => {
+                var jobs = francoisApi.getTemplateJobs(this.templateName);
+
+                jobs
+                    .map(r => r.json())
+                    .flatMap(r => r)
+                    .filter(r => r.jobName == this.jobName)
+                    .subscribe(job => {
+                        var merged = _.map(template, templateValue => this.merge(job.config.parameters, templateValue);
+
+                        this.parameters = merged;
+
+                        setTimeout(() => {
+                            $(document).foundation();
+                            this.isCreateable = true;
+                        }, 30);
+                    });
+            });
+    }
+
+    merge(jobParameters, templateValue) {
+        var jobParameter = _.findWhere(jobParameters, {name: templateValue.name});
+
+        if (jobParameter === undefined) {
+            return templateValue;
+        }
+
+        var cloned = _.clone(jobParameter);
+
+        cloned.defaultValue = templateValue.defaultValue;
+
+        return cloned;
+    }
+
+    saveJob() {
+        console.log(this.parameters);
+        this.isCreateable = false;
```

Comment:
Lol, I'll fix it in in another branch. I want to clean up all this ts since it's a tad unweildy